Workflow steps are not run in sequence deterministically

### Describe the bug

Workflow builder isn't maintaining the sequence of steps as provided. Internally it maintains a map, and therefore the sequence of steps are lost

### Describe the expected behavior

Workflow builder should ensure the sequence of steps are preserved.

### To Reproduce

The below unit test fails sometimes a Go map doesn't guarantee the ordering:
```
func TestWorkFlowBuilder_MaintainsStepSequence(t *testing.T) {
	b1 := NewStepBuilder("z", WithOnExecute(func(ctx context.Context) (*Report, error) { return &Report{}, nil }))
	b2 := NewStepBuilder("a", WithOnExecute(func(ctx context.Context) (*Report, error) { return &Report{}, nil }))
	b3 := NewStepBuilder("y", WithOnExecute(func(ctx context.Context) (*Report, error) { return &Report{}, nil }))
	b4 := NewStepBuilder("c", WithOnExecute(func(ctx context.Context) (*Report, error) { return &Report{}, nil }))
	b5 := NewStepBuilder("x", WithOnExecute(func(ctx context.Context) (*Report, error) { return &Report{}, nil }))

	wb := NewWorkFlowBuilder("wf").(*workflowBuilder)
	wb.Steps(b1, b2, b3, b4, b5)
	wf, err := wb.Build()
	require.NoError(t, err)
	require.NotNil(t, wf)

	steps := wf.(*workflow).steps
	require.Len(t, steps, 5)
	assert.Equal(t, "z", steps[0].Id())
	assert.Equal(t, "a", steps[1].Id())
	assert.Equal(t, "y", steps[2].Id())
	assert.Equal(t, "c", steps[3].Id())
	assert.Equal(t, "x", steps[4].Id())
}
```

Example output:
```
=== RUN   TestWorkFlowBuilder_MaintainsStepSequence
    workflow_builder_test.go:122: 
        	Error Trace:	/Users/leninmehedy/code/src/github.com/automa-saga/automa/workflow_builder_test.go:122
        	Error:      	Not equal: 
        	            	expected: "z"
        	            	actual  : "a"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-z
        	            	+a
        	Test:       	TestWorkFlowBuilder_MaintainsStepSequence
    workflow_builder_test.go:123: 
        	Error Trace:	/Users/leninmehedy/code/src/github.com/automa-saga/automa/workflow_builder_test.go:123
        	Error:      	Not equal: 
        	            	expected: "a"
        	            	actual  : "y"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-a
        	            	+y
        	Test:       	TestWorkFlowBuilder_MaintainsStepSequence
    workflow_builder_test.go:124: 
        	Error Trace:	/Users/leninmehedy/code/src/github.com/automa-saga/automa/workflow_builder_test.go:124
        	Error:      	Not equal: 
        	            	expected: "y"
        	            	actual  : "c"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-y
        	            	+c
        	Test:       	TestWorkFlowBuilder_MaintainsStepSequence
    workflow_builder_test.go:125: 
        	Error Trace:	/Users/leninmehedy/code/src/github.com/automa-saga/automa/workflow_builder_test.go:125
        	Error:      	Not equal: 
        	            	expected: "c"
        	            	actual  : "x"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-c
        	            	+x
        	Test:       	TestWorkFlowBuilder_MaintainsStepSequence
    workflow_builder_test.go:126: 
        	Error Trace:	/Users/leninmehedy/code/src/github.com/automa-saga/automa/workflow_builder_test.go:126
        	Error:      	Not equal: 
        	            	expected: "x"
        	            	actual  : "z"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-x
        	            	+z
        	Test:       	TestWorkFlowBuilder_MaintainsStepSequence
--- FAIL: TestWorkFlowBuilder_MaintainsStepSequence (0.00s)

Expected :x
Actual   :z
<Click to see difference>


FAIL

Process finished with the exit code 1
```

### Additional Context

_No response_

--- a/workflow_builder.go
+++ b/workflow_builder.go
@@ -12,6 +12,7 @@ type workflowBuilder struct {
 	registry     Registry
 	rollbackMode TypeRollbackMode
 	logger       zerolog.Logger
+	stepSequence []string
 	stepBuilders map[string]Builder
 	mu           sync.Mutex
 }
@@ -25,7 +26,12 @@ func (wb *workflowBuilder) Build() (Step, error) {
 	defer wb.mu.Unlock()
 
 	steps := make([]Step, 0, len(wb.stepBuilders))
-	for _, builder := range wb.stepBuilders {
+	for _, stepId := range wb.stepSequence {
+		builder, exists := wb.stepBuilders[stepId]
+		if !exists {
+			return nil, fmt.Errorf("step with id '%s' not found in builders map", stepId)
+		}
+
 		step, err := builder.Build()
 		if err != nil {
 			return nil, IllegalArgument.New("failed to build step '%s': %v", builder.Id(), err)
@@ -47,6 +53,7 @@ func (wb *workflowBuilder) Steps(steps ...Builder) WorkFlowBuilder {
 			continue
 		}
 		wb.stepBuilders[step.Id()] = step
+		wb.stepSequence = append(wb.stepSequence, step.Id())
 	}
 	return wb
 }
@@ -69,6 +76,7 @@ func (wb *workflowBuilder) NamedSteps(stepIds ...string) WorkFlowBuilder {
 			continue
 		}
 		wb.stepBuilders[id] = builder
+		wb.stepSequence = append(wb.stepSequence, id)
 	}
 	return wb
 }
@@ -122,5 +130,6 @@ func NewWorkFlowBuilder(id string) WorkFlowBuilder {
 		rollbackMode: RollbackModeContinueOnError,
 		logger:       zerolog.Nop(),
 		stepBuilders: make(map[string]Builder),
+		stepSequence: []string{},
 	}
 }

--- a/workflow_builder_test.go
+++ b/workflow_builder_test.go
@@ -1,6 +1,7 @@
 package automa
 
 import (
+	"context"
 	"errors"
 	"github.com/stretchr/testify/require"
 	"testing"
@@ -99,4 +100,26 @@ func TestWorkFlowBuilder_WithLogger_WithRollbackMode_WithRegistry(t *testing.T) 
 	assert.Equal(t, logger, wb.logger)
 	assert.Equal(t, RollbackModeStopOnError, wb.rollbackMode)
 	assert.Equal(t, reg, wb.registry)
+}
+
+func TestWorkFlowBuilder_MaintainsStepSequence(t *testing.T) {
+	b1 := NewStepBuilder("z", WithOnExecute(func(ctx context.Context) (*Report, error) { return &Report{}, nil }))
+	b2 := NewStepBuilder("a", WithOnExecute(func(ctx context.Context) (*Report, error) { return &Report{}, nil }))
+	b3 := NewStepBuilder("y", WithOnExecute(func(ctx context.Context) (*Report, error) { return &Report{}, nil }))
+	b4 := NewStepBuilder("c", WithOnExecute(func(ctx context.Context) (*Report, error) { return &Report{}, nil }))
+	b5 := NewStepBuilder("x", WithOnExecute(func(ctx context.Context) (*Report, error) { return &Report{}, nil }))
+
+	wb := NewWorkFlowBuilder("wf").(*workflowBuilder)
+	wb.Steps(b1, b2, b3, b4, b5)
+	wf, err := wb.Build()
+	require.NoError(t, err)
+	require.NotNil(t, wf)
+
+	steps := wf.(*workflow).steps
+	require.Len(t, steps, 5)
+	assert.Equal(t, "z", steps[0].Id())
+	assert.Equal(t, "a", steps[1].Id())
+	assert.Equal(t, "y", steps[2].Id())
+	assert.Equal(t, "c", steps[3].Id())
+	assert.Equal(t, "x", steps[4].Id())
 }

